jcl/src/java.base/share/classes/java/security/AccessControlContext.java

```diff
@@ -353,6 +353,41 @@ public AccessControlContext(ProtectionDomain[] fromContext) {
 	this.containPrivilegedContext = true;
 }
 
+AccessControlContext(ProtectionDomain[] pdArray, AccessControlContext parentAcc, AccessControlContext context, int authorizeState) {
+	super();
+	switch (authorizeState) {
+	default:
+		// authorizeState can't be STATE_UNKNOWN, callerPD always is NULL
+		throw new IllegalArgumentException();
+	case STATE_AUTHORIZED:
+		if (null != context) {
+			if (parentAcc == null) {
+				// inherit the domain combiner when authorized
+				this.domainCombiner = context.domainCombiner;
+			} else {
+				// when parent combiner is not null, use parent combiner to combine the current context
+				if (parentAcc.getCombiner() != null) {
```

Comment:
`parentAcc.getCombiner()` should be cached in a local for later use.

Reply:
Updated